driver-core/src/main/com/mongodb/binding/ClusterBinding.java

```diff
@@ -38,15 +39,30 @@
 public class ClusterBinding extends AbstractReferenceCounted implements ReadWriteBinding {
     private final Cluster cluster;
     private final ReadPreference readPreference;
+    private final ReadConcern readConcern;
 
     /**
      * Creates an instance.
      * @param cluster        a non-null Cluster which will be used to select a server to bind to
      * @param readPreference a non-null ReadPreference for read operations
+     * @deprecated Prefer {@link #ClusterBinding(Cluster, ReadPreference, ReadConcern)}
      */
+    @Deprecated
     public ClusterBinding(final Cluster cluster, final ReadPreference readPreference) {
+        this(cluster, readPreference, ReadConcern.DEFAULT);
+    }
+
+    /**
+     * Creates an instance.
+     * @param cluster        a non-null Cluster which will be used to select a server to bind to
+     * @param readPreference a non-null ReadPreference for read operations
+     * @param readConcern    a non-null read concern
+     * @since 3.8
+     */
+    public ClusterBinding(final Cluster cluster, final ReadPreference readPreference, final ReadConcern readConcern) {
```

Comment:
Read concern is now a property of the session context, so need to pass it in here.